hint.cabal

```diff
@@ -92,6 +92,10 @@ library
   hs-source-dirs: src
 
   ghc-options: -Wall
+
+  if impl(ghc == 8.0.1)
+    cpp-options: -DGHC_801
```

Comment:
For consistency with the rest of the code, could you use `#if __GLASGOW_HASKELL__ < 802` instead of introducing a new CPP flag from the cabal file?

Reply:
My understanding that 8.2 is 802: https://downloads.haskell.org/~ghc/7.0.3/docs/html/users_guide/version-numbering.html

Reply:
Ah, ignore me then. Should still work OK.

Reply:
> My understanding that 8.2 is 802

Huh, good catch! I have now looked at my hint commits, and it doesn't look like I have introduced any `#if __GLASGOW_HASKELL__` checks, so that misunderstanding hasn't lead to any bugs from my side at least. Thanks for the clarification!

Reply:
No problem! I will make the change as @mvdan suggests, for tidyness. I'll also make it compile!